### PR TITLE
CMR-9154 - remove tokens from /access-control/current-sids logs

### PR DESCRIPTION
**Pull request by jceaser**
Remove the token from the URL because it is showing up in /access-control/current-sids, by moving the API call from GET to POST and sending the token as data. Also added a configuration to disable the GET interface.

### CODE REVIEW

File: access-control-app/src/cmr/access_control/api/routes.clj [hunk 1]
```diff
@@ -247,141 +247,152 @@
...
 
       (context "/current-sids" []
         (OPTIONS "/" [] (common-routes/options-response))
 
-        (GET "/" {:keys [request-context params]}
-             (get-current-sids request-context params)))
+        (GET "/"
+          {:keys [request-context params]}
+
+          (when (access-control-config/enable-sids-get)
```
Comment (jceaser) [inline access-control-app/src/cmr/access_control/api/routes.clj:386]: move this up to enclose get

Comment (jceaser) [reply access-control-app/src/cmr/access_control/api/routes.clj:386]: also when is a do

File: access-control-app/src/cmr/access_control/api/routes.clj [hunk 2]
```diff
@@ -380,8 +380,20 @@
...
-        (GET "/" {:keys [request-context params]}
-             (get-current-sids request-context params)))
+        (if (access-control-config/enable-sids-get)
+          (GET "/"
+               {:keys [request-context params]}
+               (do
+                 (error (format "client [%s] Using GET instead of POST when requesting current sids"
+                                (:client-id request-context)))
+                 (get-current-sids request-context params)))
+          {:status 404})
```
Comment (jceaser) [inline access-control-app/src/cmr/access_control/api/routes.clj:390]: I tried to return 410 but ring turned it into a 404, so I gave up. This worked better then trying to do just a when

**Review by zimzoom — APPROVED** [review]

File: access-control-app/src/cmr/access_control/api/routes.clj [hunk 3]
```diff
@@ -380,8 +380,20 @@
...
+
+        (POST "/"
+              {:keys [request-context body]}
+              (get-current-sids request-context (-> (slurp body)
+                                                    (json/parse-string true)
+                                                    (clojure.walk/keywordize-keys)))))
```
Comment (eudoroolivares2016) [inline access-control-app/src/cmr/access_control/api/routes.clj:396]: nit;
```suggestion
                                                    (walk/keywordize-keys)))))
```

Comment (eudoroolivares2016) [reply access-control-app/src/cmr/access_control/api/routes.clj:396]: Then add `[clojure.walk :as walk]` as dependency

**Review by eudoroolivares2016 — APPROVED** [review]

File: access-control-app/src/cmr/access_control/api/routes.clj [hunk 4]
```diff
@@ -380,8 +380,20 @@
...
+               (do
+                 (error (format "client [%s] Using GET instead of POST when requesting current sids"
+                                (:client-id request-context)))
+                 (get-current-sids request-context params)))
+          {:status 404})
+
+        (POST "/"
+              {:keys [request-context body]}
+              (get-current-sids request-context (-> (slurp body)
+                                                    (json/parse-string true)
```
Comment (jaybarra) [inline access-control-app/src/cmr/access_control/api/routes.clj:395]: Is keywordize-keys necessary if you use `json/parse-string true`?

Comment (jceaser) [reply access-control-app/src/cmr/access_control/api/routes.clj:395]: it was when I was developing, but I'll test again

File: access-control-app/src/cmr/access_control/api/routes.clj (hunk 3 above)
Comment (jceaser) [reply access-control-app/src/cmr/access_control/api/routes.clj]: you know, I was not convinced that was going to be the solution so I didn't bother to add it, then I forgot. I ended up removing it.

File: access-control-app/src/cmr/access_control/api/routes.clj (hunk 4 above)
Comment (jceaser) [reply access-control-app/src/cmr/access_control/api/routes.clj]: removed

**Pull request merged**